descqa/DeltaSigmaTest.py

```diff
@@ -0,0 +1,118 @@
+        self.angular_diameter_distance = interp1d(z, d1, kind='quadratic')
+
+        d2 = WMAP7.comoving_transverse_distance(z) # in Mpc
+        self.comoving_transverse_distance = interp1d(z, d2, kind='quadratic')
+
+    def run_on_single_catalog(self, catalog_instance, catalog_name, output_dir):
+        # pylint: disable=no-member
+
+        res = catalog_instance.get_quantities(['redshift_true', 'ra', 'dec', 'shear_1', 'shear_2',
+                                               'convergence', 'mag_true_i_sdss', 'mag_true_z_sdss',
+                                               'mag_true_g_sdss', 'mag_true_r_sdss'])
+
+        # Compute mask for lowz sample
+        cperp = (res['mag_true_r_sdss'] - res['mag_true_i_sdss']) - (res['mag_true_g_sdss'] - res['mag_true_r_sdss'])/4.0 - 0.18
+        cpar = 0.7*(res['mag_true_g_sdss'] - res['mag_true_r_sdss']) + 1.2*((res['mag_true_r_sdss'] - res['mag_true_i_sdss'])-0.18)
+
+        mask_lowz = np.abs(cperp) < 0.2 # color boundaries
+        mask_lowz &= res['mag_true_r_sdss'] < (13.5 + cpar/0.3) # sliding magnitude cut
+        mask_lowz &= (res['mag_true_r_sdss'] > 16) &(res['mag_true_r_sdss'] < 19.6)
+
+        # Counting the number density of LOWZ galaxies
+        nlens = len(np.where(mask_lowz)[0]) / catalog_instance.sky_area
+        with open(os.path.join(output_dir, 'galaxy_density.dat'), 'a') as f:
+            f.write('{} \n'.format(nlens))
+
+        #  Additional redshift cuts used in Singh et al. (2015)
+        mask_lowz &= (res['redshift_true'] > 0.16) & (res['redshift_true'] < 0.36)
+
+        # Computing mask for source sample, this only serves to keep the number
+        # of neighbours manageable
+        mask_source = (res['mag_true_i_sdss'] < 26.) & (res['redshift_true'] > 0.4)
+
+        # Create astropy coordinate objects
+        coords = SkyCoord(ra=res['ra']*u.degree, dec=res['dec']*u.degree)
+        coords_l = coords[mask_lowz]
+        coords_s = coords[mask_source]
+
+        # Search for neighbours
+        idx1, idx2, sep2d, _ = search_around_sky(coords_l, coords_s, 2.*u.deg)
+
+        # Computing sigma crit for each pair
+        zl = res['redshift_true'][mask_lowz][idx1]
+        zs = res['redshift_true'][mask_source][idx2]
+
+        # Warning: this assumes a flat universe
+        # See http://docs.astropy.org/en/v0.3/_modules/astropy/cosmology/core.html#FLRW.angular_diameter_distance_z1z2
+        dm1 = self.comoving_transverse_distance(zl)
+        dm2 = self.comoving_transverse_distance(zs)
+        angular_diameter_distance_z1z2 = u.Quantity((dm2 - dm1)/(1. + zs), u.Mpc)
+
+        sigcrit = cst.c**2 / (4.*np.pi*cst.G) * self.angular_diameter_distance(zs) / \
+                ((1. + zl)**2. * angular_diameter_distance_z1z2 * self.angular_diameter_distance(zl))
+        # Apply unit conversion to obtain sigma crit in h Msol /pc^2
+        cms = u.Msun / u.pc**2
+        sigcrit = sigcrit*(u.kg/(u.Mpc* u.m)).to(cms) / 0.7
```

Comment:
Hard-coded 0.7's here and below are h's?  I recommend defining `h` and then using it in these equations, for future readability.

Reply:
I think `h` should be read in from the catalog's cosmology.

Reply:
Sure, a hard coded `h` is pretty ugly, shame on me... but @yymao reading h from the catalog cosmology would probably mean the whole cosmology should be consistent with the catalog's, no? In the sense that I shouldn't use the default WMAP7 (even if I think that's the right one for DC2)

Reply:
Well, I am not sure what's the best way here, because the catalogs usually won't be in the same cosmology as used in the analysis of the validation dataset. So I guess the answer depends on exactly what you want to compare. If you can compare after you scale little h out, then I guess you'll use catalogs' `h` to scale catalog data and use whatever Singh2015 uses to scale validation data.

Reply:
@yymao I've changed the code so that it will try to read and use the catalog cosmology, and only if that's not defined will it default to WMAP7. sounds good ?